app.js

```diff
@@ -0,0 +1,124 @@
+    Dê um nome para a empresa e um telefone fictício, preechendo essas informações
+    no arquivo company.json que já está criado.
+
+    Essas informações devem ser adicionadas no HTML via Ajax.
+
+    Parte técnica:
+    Separe o nosso módulo de DOM criado nas últimas aulas em
+    um arquivo DOM.js.
+
+    E aqui nesse arquivo, faça a lógica para cadastrar os carros, em um módulo
+    que será nomeado de "app".
+    */
+    var app = (function appControler() {
+        return {
+
+            init: function() {
+                this.companyInfo();
+                this.initEvents();
+            },
+
+            initEvents: function initEvents() {
+                $('[data-js="form-register"]').on('submit', this.handleSubmit);
+            },
+
+            handleSubmit: function handleSubmit(e) {
+                e.preventDefault();
+                var $tableCar = $('[data-js="table-car"]').get();
+                $tableCar.appendChild(app.createNewCar());
+            },
+
+            createNewCar: function createNewCar() {
+                var $fragment = document.createDocumentFragment(),
+                    $tr = document.createElement('tr'),
+                    $tdImage = document.createElement('td'),
+                    $imagem = document.createElement('img'),
+                    $tdCor = document.createElement('td'),
+                    $tdAno = document.createElement('td'),
+                    $tdBrandModel = document.createElement('td'),
+                    $tdPlate = document.createElement('td'),
+                    $tdRemove = document.createElement('td'),
+                    $btnRemove = document.createElement('button'),
+                    imagemUrl = $('[data-js="image"]').get().value;
+
+                $tdBrandModel.textContent = $('[data-js="brand-model"]').get().value;
+                $tdCor.textContent = $('[data-js="color"]').get().value;
+                $tdAno.textContent = $('[data-js="year"]').get().value;
+                $tdPlate.textContent = $('[data-js="plate"]').get().value;
+                $btnRemove.textContent = 'Remover';
+                $btnRemove.setAttribute('data-js', 'btn-remove');
+
+                $imagem.setAttribute('src', imagemUrl)
+                $tdImage.appendChild($imagem);
+                $tr.appendChild($tdImage);
+                $tr.appendChild($tdBrandModel);
+                $tr.appendChild($tdAno);
+                $tr.appendChild($tdPlate);
+                $tr.appendChild($tdCor);
+                $tdRemove.appendChild($btnRemove);
+                $tr.appendChild($tdRemove);
+
+                $btnRemove.addEventListener('click', this.removeRegister, false);
+                //$('[data-js="btn-remove"]').on('click', this.removeRegister);
+
+                return $fragment.appendChild($tr);
+            },
+
+            removeRegister: function removeRegister(e) {
+                this.parentNode.parentNode.remove();
```

Comment:
Fala @gustavom! O problema na verdade está na forma de remover o elemento! Provavelmente o `this` aqui não é quem você esperava que fosse =)

O ideal, para evitar esse tipo de problema, é tentar evitar ao máximo o uso do `this`. No caso dessa aplicação, você pode precisar dele em alguns momentos, mas tente usar somente onde você sabe que ele não vai mudar, o que não é o caso de callbacks de evento =)

Então, pra resolver, você pode fazer o seguinte: adicione um identificador único em cada linha.. quando clicar em remover, verifique por esse identificador, e então remova a linha. Deu pra sacar? =)